stdlib/builtins.pyi

```diff
@@ -214,6 +214,8 @@ _NegativeInteger: TypeAlias = Literal[-1, -2, -3, -4, -5, -6, -7, -8, -9, -10, -
 _LiteralInteger = _PositiveInteger | _NegativeInteger | Literal[0]  # noqa: Y026  # TODO: Use TypeAlias once mypy bugs are fixed
 
 class int:
+    @overload
+    def __new__(cls) -> Literal[0]: ...
```

Comment:
Not sure if this is actually safe?

```pycon
>>> class Foo(int): ...
...
>>> type(Foo())
<class '__main__.Foo'>
```

In general our approach is to give more precise signatures for methods on classes, even if they're not so precise for subclasses. But these overloads would lead to typecheckers making inferences that are flat-out wrong when it comes to the return type of these methods on subclasses.

Reply:
The changes you're making to `bool` are probably okay, since `bool` can't be subclassed

Reply:
Oh that's fair, maybe this can be re-looked at if we get an Intersection then

Reply:
> maybe this can be re-looked at if we get an Intersection then

Thinking about a hypothetical type like `Self & Literal[0]` makes my brain go fuzzy, but sure 😆